open-metadata-implementation/common-services/ffdc-services/src/main/java/org/odpi/openmetadata/commonservices/ffdc/rest/GenericResponse.java

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.commonservices.ffdc.rest;
+
+import java.util.Collection;
+import java.util.List;
+
+
+public interface GenericResponse<R> extends FFDCResponse {
+
+    /**
+     * Add several results for the response
+     * @param results collection with results
+     * */
+    void addAllResults(Collection<? extends R> results);
+
+    /**
+     * Add single result for the response
+     * @param result - one result
+     * */
+    void addResult(R result);
+
+    /**
+     * Get all results of the response
+     *
+     * @return results
+     **/
+    List<R> getResult();
+
+    /**
+     * Get head element from result array.
```

Comment:
I suggest not using getHead as the method name. I assume this refers to the head (first) element in the array, which is terminology referring to the implementation. I suggest changing this to getOnlyResult()- or similar-  which makes more sense for the caller.     
This led me to think that it might be clearer to have 2 child interfaces 
* GenericMultiResponse - with getResults method 
* GenericSingleResponse - with getResult method

The caller then uses the appropriate interface and associated method.

Reply:
In this case, this is just an auxiliary method, for those cases when we clearly know that we have a single result in our response. I don't think that we should divide into 2 interfaces in this case, it's just an auxiliary method for more convenient work on the back end and nothing more.